code/game/objects/items/devices/taperecorder.dm

```diff
@@ -287,4 +602,55 @@
 //Random colour tapes
 /obj/item/device/tape/random/New()
 	icon_state = "tape_[pick("white", "blue", "red", "yellow", "purple")]"
-	..()
\ No newline at end of file
+	..()
+
+/*
+// A constructor for map-makers to make tapes with messages
+/obj/item/device/tape/proc/New(txt, color="white")
+	icon_state = "tape_[color]"
+	// setDialogue(txt) // FIXME
+	premade = TRUE
+	..()
+	*/
+
+// FIXME: change back to proc
+/obj/item/device/tape/verb/setDialogue(txt as text)
+	premade = TRUE
+	// Reset everything
+	used_capacity = 1
+	max_capacity = 600
+	storedinfo = list()
+	timestampInSeconds = list()
+
+	var/start = 1
+	// For reference: https://regex101.com/r/0jjDGb/6
+	// Our regex: \[([^\]]*)\]([^[]*)
+	var str = "\\\[(\[^\\]]*)\\](\[^\[]*)"
+	var/regex/re = new(str,"g")
+	var/regex/re2 = new("^(.*);$")
+	var/regex/re3 = new("^wait=(\\d+)$")
+	while(re.Find(txt, start))
+		start = re.next // Get our next match
+		var/speaker = ""
+		var/message = ""
+		var/wait = 0
+		if (re3.Find(re.group[1])) // Is this a [wait] command?
+			wait = re3.group[1] // Add silence
+			used_capacity += round(text2num(wait))
+		else
+			if (length(re.group[1]) != 0) // Is it a blank []?
+				if (re.group[1] == ";") // Quick announcement?
+					speaker = TAPE_RECORDER_NAME

```

Comment:
I wanted to access the name of /obj/item/device/taperecorder from /obj/item/device/tape/verb/setDialogue(txt as text).
/obj/item/device/taperecorder has a variable of an instance of /obj/item/device/tap, so I thought the easiest way would be to use a #define without having to initialize a /obj/item/device/taperecorder for its name variable.